src/gui/MainWindow.cpp

```diff
@@ -76,6 +94,9 @@ MainWindow::MainWindow() :
 	m_metronomeToggle( 0 ),
 	m_session( Normal )
 {
+#ifdef LMMS_BUILD_LINUX
```

Comment:
Perhaps a `#ifndef` Mac and Windows check would be sufficient here instead to avoid non-Linux KDEs from being impacted (e.g. BSD, etc).

Reply:
My concern there is that it would be basically untested. The CMake file change would also have to be set up this way as well.

Reply:
What do you mean by "untested"?

Reply:
CI doesn't cover platforms other than windows, mac, and linux, and I don't have a machine to attempt building for anything else. This code is very much not portable, and if allowances are being for other platforms the code should be compiled for them at some point.

Reply:
@BumblingCoder perhaps there's some confusion... I'm asking you disable the option for Mac + Windows, which will inadvertently enable it for all others.  Enabling it for Linux-only is going to just cause this same issue to happen for (e.g.) OpenBSD users that chose the KDE desktop.  Regardless, it's a safe call for all platforms everywhere since the library won't be found, I was just trying to be consistent with the "probably only on KDE desktop" logic that I presume you were choosing.

Reply:
There is stuff in the CMake files about Haiku. Is this something that we intend to support? I would be more ok making this change say linux or bsd than saying not windows or mac.

Reply:
>  I would be more ok making this change say linux or bsd than saying not windows or mac.

Well Haiku isn't BSD, it's BeOS and it's an edge-case (and it can't run the full KDE desktop yet apparently).  However, most Unixes can run KDE.

I'd rather blacklist than whitelist, it's more scalable.  KDE will never be a primary desktop on Windows or MacOS in the foreseeable future.  I can't say the same for new/undocumented Unixes.

Reply:
My concern isn't really about where KDE is common, but about putting in stuff depending on dlopen() and specifying where to run that by specifying the platforms it isn't available or needed on rather than the ones it is. I could go either way on this though.

Reply:
> putting in stuff depending on `dlopen()` and specifying where to run that by specifying the platforms it isn't available or needed

Great point.  As BeOS seems to be the major exception from what we support.  Even [Solaris supports `dlopen()`](https://docs.oracle.com/cd/E23824_01/html/821-1465/dlopen-3c.html).   Until we require `dlopen` for something else, I would suggest simply blacklisting Haiku along side Windows and Mac then.  Down the road if we need `dlopen` for a future feature, we can [leverage a cross-platform wapper](https://www.gnu.org/software/libtool/manual/html_node/Using-libltdl.html).